github/timestamp.go

```diff
@@ -29,6 +29,9 @@ func (t *Timestamp) UnmarshalJSON(data []byte) (err error) {
 	i, err := strconv.ParseInt(str, 10, 64)
 	if err == nil {
 		t.Time = time.Unix(i, 0)
+		if t.Time.After(time.Now()) {
```

Comment:
My main concern with this implementation revolves around the following question:

Are we guaranteed that _all_ timestamps sent from GitHub will always be in the past?

I think the answer is "no"... I could imagine that a timestamp might represent when something expires in the future.  (There's also the matter of time skew between GitHub's clocks and the user's clocks that makes this fragile for "close-to-now" timestamps.)

I wrote a little experiment locally:

```
go build parsetime.go && parsetime in five minutes
"in five minutes" in milliseconds since epoch is:
1617717693956
Tuesday, 06-Apr-21 10:01:33 EDT
Tuesday, 06-Apr-21 14:01:33 UTC
Interpreting this as if it were seconds returns: 53233-06-07 06:05:56 -0400 EDT
```

and found out that interpreting milliseconds since epoch as seconds since epoch for a "now" timestamp returns the year 53233...

So I was thinking that a better approach would be to first parse the time as seconds, but then if the year is something greater than, say, 3000, go ahead and interpret it as milliseconds since epoch.

How does that sound to you?

Reply:
It's amazing what you can't come up with at midnight. When I was writing this I was trying to think of scenarios where there might be a timestamp in the future, but couldn't think of one. I was also concerned about time "wobble" with timestamps generated from events occurring ridiculously close to the users clock and originally added a single day to the current date, but decided to back it out.

But I think just checking if the year is greater than 3,000 is elegantly simple enough, and hopefully software progresses enough in the next 1,000 years that this is obsolete by then 😂 

Updated to check for a `year > 3000`